File: ingestion-service/internal/middleware/internal_auth.go
package middleware

import "github.com/gofiber/fiber/v2"

// InternalAuth는 X-Internal-Token 헤더를 검증하는 Fiber 미들웨어다.
// /health 경로는 검증에서 제외한다.
func InternalAuth(expectedToken string) fiber.Handler {
	return func(c *fiber.Ctx) error {
		if c.Path() == "/health" {
			return c.Next()
		}
		token := c.Get("X-Internal-Token")
		if token != expectedToken {
			return c.Status(fiber.StatusForbidden).JSON(fiber.Map{
				"error": "invalid internal token",
			})
		}
		return c.Next()
	}
}

File: query-service/src/middleware/internal-auth.ts
import type { MiddlewareHandler } from "hono";

export function internalAuth(expectedToken: string): MiddlewareHandler {
  return async (c, next) => {
    if (c.req.path === "/health") {
      await next();
      return;
    }
    const token = c.req.header("x-internal-token");
    if (token !== expectedToken) {
      return c.json({ error: "invalid internal token" }, 403);
    }
    await next();
  };
}
